index.js

```diff
@@ -99,13 +144,71 @@ function createRenderer() {
 }
 
 /**
+ * Get service iframe video by passed name
+ * Required href:
+ * - youtube: https://youtu.be/1GWoMnYldYc or https://www.youtube.com/watch?v=1GWoMnYldYc
+ * - yandex: https://video.yandex.ru/iframe/ya-events/clg24o8tu3.7046/
+ * - vimeo: https://vimeo.com/30838628 or https://player.vimeo.com/video/30838628
+ * @param {String} href - full video href, examples see above
+ * @param {String} serviceName - name of the service [youtube, yandex, vimeo]
+ * @returns {String} - iframe video block
+ */
+function getIframe(href, serviceName) {
+    var arr = href.split('/'),
+        key = arr.pop(),
+        iFrame;
+
+    /**
+     * Get the second last href arr item for key,
+     * if href contain slash at the end and our key empty
+     * For example: https://vimeo.com/53219242/
+     */
+    if (!key.length) {
+        key = arr[arr.length - 1];
+    }
+
+    /**
+     * Key contain extra words
+     * For example:
+     * Original href = http://www.youtube.com/watch?v=1GWoMnYldYc
+     * key = watch?v=1GWoMnYldYc - we need remove watch?v=
+     */
+    if (serviceName === 'youtube' && key.indexOf('watch?v=') > -1) {
+        key = key.match(/(\d\w).+$/gi)[0];
+    }
+
+    iFrame = {
+        youtube: [
+            '<iframe src="//www.youtube.com/embed/',
+            key,
+            '" width="560" height="315" frameborder="0" allowfullscreen></iframe>'
+        ],
+        yandex: [
+            '<iframe src="//video.yandex.ru/iframe/ya-events/',
+            key,
+            '" width="560" height="315" frameborder="0"></iframe>'
+        ],
+        vimeo: [
+            '<iframe src="//player.vimeo.com/video/',
+            key,
+            '" width="560" height="315" frameborder="0" webkitallowfullscreen mozallowfullscreen ',
+            'allowfullscreen></iframe>'
+        ]
+    };
+
+    return iFrame[serviceName].join('');
+}
+
+/**
  * Return an instance of custom marked renderer
  * Reset usedAnchors variable,
  * which need to check duplicate headers in the markdown
  * @returns {*}
  */
 function getRenderer() {
-    if (!renderer) renderer = createRenderer();
+    if (!renderer) {
+        renderer = createRenderer();
+    }
```

Comment:
@gela-d it won't work without brackets: `renderer || (renderer = createRenderer())`